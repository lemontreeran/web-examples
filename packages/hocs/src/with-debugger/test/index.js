import React from 'react'
import { mount } from 'enzyme'

import withDebugger from '../src/'

const Target = () => null

describe('withDebugger', () => {
  describe('env', () => {
    const origNodeEnv = process.env.NODE_ENV

    afterAll(() => {
      process.env.NODE_ENV = origNodeEnv
    })

    it('should pass Target through in production env', () => {
      process.env.NODE_ENV = 'production'

      const EnhancedTarget = withDebugger(Target)
      const wrapper = mount(
        <EnhancedTarget />
      )

      expect(wrapper).toMatchSnapshot()
    })

    it('should wrap display name in non-production env', () => {
      process.env.NODE_ENV = 'test'

      const EnhancedTarget = withDebugger(Target)
      const wrapper = mount(
        <EnhancedTarget />
      )

      expect(wrapper).toMatchSnapshot()
    })
  })
})
