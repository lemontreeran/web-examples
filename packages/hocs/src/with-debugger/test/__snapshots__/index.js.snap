// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withDebugger env should pass Target through in production env 1`] = `<Target />`;

exports[`withDebugger env should wrap display name in non-production env 1`] = `
<withDebugger(Target)>
  <Target />
</withDebugger(Target)>
`;
